crates/js-component-bindgen/src/intrinsics.rs

```diff
@@ -96,6 +97,19 @@ pub fn render_intrinsics(
                 const hasOwnProperty = Object.prototype.hasOwnProperty;
             "),
 
+            Intrinsic::ToResultString => output.push_str("
+                function toResultString(obj) {
+                    return JSON.stringify(obj, (_, v) => {
+                        if (v instanceof Uint8Array) {
+                            return `[Uint8Array (${v.byteLength})]`;
+                        } else if (typeof v === 'bigint') {
```

Comment:
We do also support other data array type lifting in returns, so I think this should be something more like:

```suggestion
                        if (v instanceof Uint8Array.prototype) {
                            return `[${v[Symbol.toStringTag]} (${v.byteLength})]`;
                        } else if (typeof v === 'bigint') {
```